src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -50,26 +54,38 @@ public CommandResult execute(Model model) throws CommandException {
 
         Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
 
-        /*
-        this part of code should only exist in javafx file so remove it so far
-        Show confirmation dialog
-        Alert alert = new Alert(Alert.AlertType.CONFIRMATION);
-        alert.setTitle("Delete Confirmation");
-        alert.setHeaderText("Are you sure you want to delete this person?");
-        alert.setContentText(String.format("Name: %s", personToDelete.getName()));
-
-        ButtonType result = alert.showAndWait().orElse(ButtonType.CANCEL);
-        if (result != ButtonType.OK) {
-            return new CommandResult("Deletion cancelled.");
+        if (!isConfirmed) {
```

Comment:
Can control of this section possibly be passed to UI? It feels weird to handle JavaFx and GUI matters in the Logic section of the code even though we agreed mostly that MVC does not have to be strictly followed.